kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -145,6 +146,17 @@ public TransactionCommitResult commit(Engine engine, CloseableIterable<Row> data
       CommitInfo attemptCommitInfo = generateCommitAction(engine);
       updateMetadataWithICTIfRequired(
           engine, attemptCommitInfo.getInCommitTimestamp(), readSnapshot.getVersion(engine));
+
+      // If row tracking is supported, assign base row IDs and default row commit versions to any
+      // AddFile or RemoveFile actions that do not yet have them. If the row ID high watermark
+      // changes, emit a DomainMetadata action.
+      Optional<DomainMetadata> highWaterMark =
```

Comment:
Yes, this would be better. I've updated this to
```java
if (TableFeatures.isRowTrackingSupported(protocol)) { /* RowTracking codes */ }
```
, and `RowTracking` methods can believe they will only ever be called if it is supported.